### ISSUE TITLE
Interface conversion error with new version

### ISSUE THREAD

**Issue by danitrod** [antonmedv/expr#422]
I'm trying to upgrade `expr` version from `v1.9.0` to `v1.14.3`, and the following expression, that before would compile, now fails:
```
error evaluating expression: interface conversion: interface {} is float64, not int (1:8)
 | 5 ** 3 == 150
 | .......^
```

**Comment by antonmedv** [comment]
I think I got the problem: 

The expression `5 ** 3 == 150` is compiled to 
```
0  OpPush  <0>  125.0
1  OpPush  <1>  150
2  OpEqualInt
```

But `OpEqualInt` is not designed to compare float64 and int.

So it must be type checker and optimizer doing wrong stuff. 

For example in `.5 == 1` compiled program is correct: 

```
0  OpPush  <0>  0.5
1  OpPush  <1>  1
2  OpEqual
```

The `OpEqual` is the right opcode. Will fix the bug.

**Comment by antonmedv** [comment]
Fixed in 4629871ec39fffeb8533b932927c8e9f362b8b55
